inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+ *        This process shall be repeated in a loop until the totality of the `Data Source` has been
+ *        copied.
+ *      - If there is no more data to return, the flush shall return #AZ_ULIB_EOF and an empty
+ *        `const az_span* const` shall be created and passed to the `push_callback`.
+ * 
+ * @param[in]       ustream_forward_instance    The #az_ulib_ustream_forward* with the interface of
+ *                                              the ustream_forward.
+ * @param[out]      push_callback               The #az_ulib_flush_callback* for the consumer to handle
+ *                                              the incoming data.
+ * @param[out]      push_callback_context       The #az_context* contract between the caller and
+ *                                              push_callback.
+ * 
+ * @pre     \p ustream_forward_instance shall not be `NULL`.
+ * @pre     \p ustream_forward_instance shall be a valid ustream that is the implemented ustream
+ *             type.
+ * @pre     \p push_callback shall not be `NULL`.
+ * @pre     \p push_callback_context shall not be `NULL`.
+ * 
+ * @return The #az_result with the result of the flush operation
+ *      @retval #AZ_OK                        If the flush operation succeeded in creating the
+ *                                            `const az_span* const` buffer and calling the
+ *                                            `push_callback`.
+ *      @retval #AZ_ULIB_EOF                  If the flush operation has no more data to grab and
+ *                                            is ready to terminate.
+ *      @retval #AZ_ERROR_ULIB_BUSY           If the resource necessary to get the next portion of
+ *                                            data is busy.
+ *      @retval #AZ_ERROR_CANCELED            If any one of the flush operation's dependent
+ *                                            external calls is canceled.
+ *      @retval #AZ_ERROR_NOT_ENOUGH_SPACE    If there is not enough memory to finish copying the
+ *                                            data from source to destination.
+ *      @retval #AZ_ERROR_ULIB_SECURITY       If any one of the flush operation's dependent
+ *                                            external calls returns an error for security reasons.
+ *      @retval #AZ_ERROR_ULIB_SYSTEM         If any one of the flush operation's dependent
+ *                                            fails at the system level.
+ */ 
+AZ_INLINE az_result az_ulib_ustream_forward_flush(
+    az_ulib_ustream_forward* ustream_forward_instance,
+    az_ulib_flush_callback push_callback, 
+    az_context* push_callback_context)
+{
+  return ustream_forward_instance->control_block->api->flush(
+            ustream_forward_instance, 
+            push_callback, 
+            push_callback_context);
+}
+
+/**
+ * @brief   Gets the next portion of the ustream_forward starting at the current position.
```

Comment:
>starting at the current position [](http://example.com/codeflow?start=57&length=32)

Consider hide any information about "current position" to the user of this API. Perhaps you can use something like "next sequential portion".

Reply:
This function must behave the same as ustream's read, so user must know what the current position in the stream is such that they can call this function in sequence with some sort of write function (just as is done with ustream's read).